src/PointwiseFunctions/AnalyticSolutions/GrMhd/SmoothFlow.hpp

```diff
@@ -58,6 +57,12 @@ class SmoothFlow : virtual public MarkAsAnalyticSolution,
              const std::array<double, 3>& wavevector, double pressure,
              double adiabatic_index, double perturbation_size);
 
+  /// \cond
+  explicit SmoothFlow(CkMigrateMessage* msg);
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(SmoothFlow);
```

Comment:
Include `CharmPupable`.

Also add virtual inheritance from `evolution::initial_data::InitialData` to this class, or is the public inheritance from RelEuler enough?

Reply:
:+1:

Not needed because of the public inheritance. It is needed for the magnetized disk and star because those use private inheritance.